dace/transformation/dataflow/redundant_array.py

```diff
@@ -49,10 +49,15 @@ def _validate_subsets(edge: graph.MultiConnectorEdge,
             desc = arrays[src_name]
             if isinstance(desc, data.View) or edge.data.data == dst_name:
                 src_subset = subsets.Range.from_array(desc)
-                if src_subset.num_elements() != dst_subset.num_elements():
+                src_expr = src_subset.num_elements()
+                src_expr_exact = src_subset.num_elements_exact()
+                dst_expr = dst_subset.num_elements()
+                dst_expr_exact = dst_subset.num_elements_exact()
+                if (src_expr != dst_expr and symbolic.inequal_symbols(
+                        src_expr_exact, dst_expr_exact)):
                     raise ValueError(
-                        "Source subset is missing (dst_subset: {}, "
-                        "src_shape: {}".format(dst_subset, desc.shape))
+                        "Destination subset is missing (src_subset: {}, "
```

Comment:
what happened here with the exception?

Reply:
If you mean about the message, it is obviously a copy-paste issue. It is exactly the same code as below, but the message should be different.

Reply:
yes

Reply:
Done